game-core/src/main/java/games/strategy/engine/chat/Chat.java

```diff
@@ -112,26 +110,22 @@ public void participantRemoved(final PlayerName playerName) {
             node -> {
               chatters.remove(node);
               updateConnections();
-              for (final IChatListener listener : listeners) {
-                listener.addStatusMessage(node.getPlayerName() + " has left");
-              }
+              chatMessageListeners.forEach(
+                  listener -> listener.addStatusMessage(node.getPlayerName() + " has left"));
             });
   }
 
   @Override
   public void slappedBy(final PlayerName from) {
     final String message = "You were slapped by " + from;
-    for (final IChatListener listener : listeners) {
-      chatHistory.add(new ChatMessage(message, from));
-      listener.addMessageWithSound(message, from, SoundPath.CLIP_CHAT_SLAP);
-    }
+    chatHistory.add(new ChatMessage(message, from));
```

Comment:
Bugfix is here, notice that 'chatHistory.add(...)' is in the for loop, yet does not use any of the loop variables. Debugging, on a single slap this is called 'n' times. This line was moved outside of the for-loop so that it is called just once.

Perhaps also worth noting that the chat history line is outside of listener for-loops in other locations as well: https://github.com/triplea-game/triplea/pull/5313/files#diff-8ffdcb54556ca7b66cd59050e599e646R84